frontend/src/helpers/endpoint.ts

```diff
@@ -1,27 +1,43 @@
-import { SERVICE } from "@/types/enums";
+"use server";
+import { HTTP_METHODS, SERVICE } from "@/types/enums";
 import { getLogger } from "./logger";
+import { revalidateTag } from "next/cache";
+
 const logger = getLogger("api");
 
-type apiConfig = {
-  method: string;
-  service: SERVICE;
-  path: string;
-  body?: {};
-  tags?: string[]; // cache scope
+/**
+ * Configuration object for API calls.
+ */
+type ApiConfig = {
+  method: HTTP_METHODS; // HTTP method.
+  service: SERVICE; // Enum representing the service type.
+  path?: string; // Optional endpoint path.
+  body?: {}; // Optional request body.
+  tags?: string[]; // Optional array of caching scopes.
 };
 
 /**
- * Production: service_api_url/<path>
- * Development: localhost:<service_ports>/<path>
- * @param service
- * @param path
+ * Response object for API calls.
  */
-export const api = async (config: apiConfig) => {
+type ApiResponse = {
+  status: number; // HTTP status code.
+  message: string;
+  data?: any; // Response data.
+};
+
+/**
+ * Handles API calls to the backend gateway.
+ * @param config {ApiConfig} - Configuration object for the API call.
+ * @returns {Promise<ApiResponse>} - Response from the API call.
+ */
+export default async function api(config: ApiConfig): Promise<ApiResponse> {
+  // Configure gateway host based on the environment (production or development).
   const host =
     process.env.NODE_ENV == "production"
       ? process.env.ENDPOINT_PROD
       : process.env.ENDPOINT_DEV;
 
+  // Configure local service port based on the 'service' property in the configuration.
   let servicePort = ":";
```

Comment:
I have created an issue [here ](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/issues/41)asking for the same question: I was curious if we need to always assign a port number before we call an endpoint. I thought we could pre-defined it and then keep using it in each service?